packages/core/src/MachineNode.ts

```diff
@@ -0,0 +1,351 @@
+import { toArray, isBuiltInEvent, toSCXMLEvent } from './utils';
+import {
+  Event,
+  StateValue,
+  MachineOptions,
+  EventObject,
+  TransitionDefinition,
+  DelayedTransitionDefinition,
+  StateSchema,
+  ActionObject,
+  Mapper,
+  PropertyMapper,
+  NullEvent,
+  MachineConfig,
+  SCXML,
+  Typestate,
+  TransitionDefinitionMap
+} from './types';
+import { State } from './State';
+
+import { toActionObject } from './actions';
+import { IS_PRODUCTION } from './environment';
+import { STATE_DELIMITER } from './constants';
+import { getConfiguration, getChildren, getAllStateNodes } from './stateUtils';
+import {
+  getStateNodeById,
+  getInitialState,
+  getStateNodes,
+  transitionNode,
+  resolveStateValue,
+  resolveTransition
+} from './nodeUtils';
+import { StateNode } from './StateNode';
+
+export const NULL_EVENT = '';
+export const STATE_IDENTIFIER = '#';
+export const WILDCARD = '*';
+
+const createDefaultOptions = <TContext>(
+  context: TContext
+): MachineOptions<TContext, any> => ({
+  actions: {},
+  guards: {},
+  services: {},
+  activities: {},
+  delays: {},
+  context
+});
+
+function resolveContext<TContext>(
+  context: TContext,
+  partialContext: Partial<TContext>
+): TContext {
+  if (context === undefined) {
+    return context;
+  }
+
+  return {
+    ...context,
+    ...partialContext
+  };
+}
+
+export class MachineNode<
+  TContext = any,
+  TStateSchema extends StateSchema = any,
+  TEvent extends EventObject = EventObject,
+  TTypestate extends Typestate<TContext> = any
+> extends StateNode<TContext, TStateSchema, TEvent> {
+  public context: TContext;
+  /**
+   * The relative key of the state node, which represents its location in the overall state value.
+   */
+  public key: string;
```

Comment:
this is already part of the StateNode, so we don't have to redefine it here

Reply:
And same can be said about many other properties defined here - on the MachineNode